skipchain/skipchain.go

```diff
@@ -68,6 +69,10 @@ type Service struct {
 // added.
 func (s *Service) StoreSkipBlock(psbd *StoreSkipBlock) (*StoreSkipBlockReply, onet.ClientError) {
 	prop := psbd.NewBlock
+	if !s.ServerIdentity().Equal(prop.Roster.Get(0)) {
```

Comment:
In fact every new block can have a completely different `Roster`, so the problem is far worse ;)
This will only be better in skipchain.v2 where there will be a unified way of handling the roster. So if there are no other comments, I think this is ready to merge.